Upgrade rspec to later version

surminus opened the pull request:
the last_comment method was removed in Rake, and this is fixed in a later version of rspec.

Ref: https://github.com/rspec/rspec-core/pull/2197

annashipman commented:
👍

surminus commented:
Gah, didn't fix it 🤔

annashipman commented:
Well it got past it on the first run and now is struggling with nokogiri... I refer you to bob...

surminus commented:
I've added a bunch more changes which _should_ fix it for all versions.

surminus commented:
I've repushed the fix and made commits more specific, and bumped the minor version.

Getting these tests fixed will help fix up the fog dependency which is a bug I've just hit when trying to deploy something 😢 

/cc @annashipman @rjw1

annashipman commented:
This looks good to me: the tests pass, well done!

I'm not super-keen on pinning Rake to a version less than 11 though - is there a future in which we can use the latest Rake?

surminus commented:
I'll investigate fixing this with a more up to date Rake.

surminus commented:
Added several more commits. The error was actually being thrown by rubocop, so we have a newer Rake and an updated rubocop, which also involved a few syntax updates.

surminus commented:
Failing on Ruby 2.0.0 🤕 

```
Failure/Error:
    SimpleCov.adapters.define 'gem' do
      add_filter '/spec/'
      add_filter '/vendor/'
  
      add_group 'Libraries', '/lib/'
    end
```

rjw1 commented:
is 2.0.0 still a supported ruby?

surminus commented:
Ruby 2.0.0 is no longer supported. We can remove this support as it breaks a lot of things.

annashipman commented on CHANGELOG.md:
```diff
@@ -1,3 +1,9 @@
+## 2.0.1 (2017-05-30)
```
Today's date instead?

annashipman commented:
This all looks good to me and well done for fixing all those errors. Let's remove support for Ruby 2.0.0. next time it causes a problem?

I've got one tiny comment on the date but other than that, 👍 merge away!

surminus commented:
Fixed the date! One thing to note is that since we moved CI the gem is no longer automatically published to rubygems, so I'll need do the work to ensure further new versions are pushed. I believe this has caused a problem with other dependencies.

surminus commented:
I manually published the new gem (not sure why I'm doing this on a Sunday):

https://rubygems.org/gems/vcloud-core/versions/2.0.1

annashipman commented:
You definitely [shouldn't be doing it on a Sunday](http://dan.carley.co/blog/2014/05/21/working-late-responsibly/) but 👍 well done for getting this finished. Thanks!
